dataland-keycloak/start_keycloak.sh

```diff
@@ -1,14 +1,29 @@
 #!/bin/bash
 set -eu
 
+mode=$1
+dataland_realm_folder=/opt/keycloak/datalandsecurity
```

Comment:
So, in the end everything is copied here and when something goes wrong, this could serve as a backup? And this will be overwritten whenever keycloak is restarted?
Should there be something that makes checkpoints that can serve as further backups? Security issues?

Reply:
The backup folder is filled and cleared (mv) in subsequent commands. So there hopefully should not be too much of a security risk. This backup folder is only relevant if initialize_keycloak.sh fails at a point where the keycloak docker volume does not exist, i.e. after the volume gets deleted and before the time out loop waiting for the keycloak-initializer to finish initialization terminates without time out. In this case the /home/ubuntu/dataland/dataland-keycloak/users persists until the next execution of start_and_deploy_to_server.sh where it gets deleted and recreated empty. That is where we use the backup folder.